zapcore/write_syncer_bench_test.go

```diff
@@ -39,7 +42,7 @@ func BenchmarkMultiWriteSyncer(b *testing.B) {
 			}
 		})
 	})
-	b.Run("4", func(b *testing.B) {
+	b.Run("4 discarder", func(b *testing.B) {
```

Comment:
This looks like a 2 discarder? Any difference between here and on line 33?

Reply:
This benchmark already exist in master branch. The idea to add this text to provide more context in the benchmark. 

Here the `4 discarder` refers that we are benchmarking `NewMultiwriterSyncer` with 4 `ztest.Discarder{}` 

With the new `BufferedWriteSyncer` there is a need to make a distinction to make sure the benchmark shows the proper information between benchmarks. 
```
BenchmarkMultiWriteSyncer/2_discarder-4         	41213204	        66.5 ns/op	      16 B/op	       1 allocs/op
BenchmarkMultiWriteSyncer/4_discarder-4         	19545279	        72.0 ns/op	      16 B/op	       1 allocs/op
BenchmarkMultiWriteSyncer/4_discarder_with_buffer-4         	 8759560	       143 ns/op	      16 B/op	       1 allocs/op
```

Reply:
Sorry, what i meant is that i dont see a difference between benchmarks on 33:44 and 45:58. They look the same to me.

Reply:
These test are practically the same just with more `ztest.Discarder{}`. For context this test exist since 2017

Reply:
derp, the last 2 are hidden in the review. sorry, it took me a minute to see that they get cut off in the code review. carry on.